root/static/scripts/annotation/AnnotationHistoryTable.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * @flow
+ * Copyright (C) 2022 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import mutate from 'mutate-cow';
+import * as React from 'react';
+
+import EditorLink from '../common/components/EditorLink';
+import bracketed from '../common/utility/bracketed';
+import formatUserDate from '../../../utility/formatUserDate';
+
+type Props = {
+  +$c: CatalystContextT,
+  +annotations: $ReadOnlyArray<AnnotationT>,
+  +baseUrl: string,
+};
+
+/* eslint-disable flowtype/sort-keys */
+type ActionT =
+  | {+type: 'update-new', +annotationId: number}
+  | {+type: 'update-old', +annotationId: number}
+;
+/* eslint-enable flowtype/sort-keys */
+
+type StateT = {
+  +selectedNew: number,
+  +selectedOld: number,
+};
+
+type WritableStateT = {
+  selectedNew: number,
+  selectedOld: number,
+};
```

Comment:
One trick (which is more helpful for types with a lot of properties) is that you can just spread the read-only type to remove all the variance/read-onlyness.
```suggestion
type WritableStateT = {...StateT}; // this has writable properties
```